### DISCUSSION TITLE
[Feature proposal] Scroll Snap

### DISCUSSION THREAD

**Discussion by jesse-deboer** [tailwindlabs/tailwindcss#1406] [Ideas]
Hey! 👋

I love using TailwindCSS and use it even for making sliders while using minimal custom css. There is just one thing I need to declare to make use of sliders on mobile that I think would be a great addition to TailwindCSS:

**Scroll snapping**
For better horizontal (could also be vertical) scrolling on mobile I use scroll-snap in the following way:

```
.scroll-x-mandatory {
	scroll-snap-type: x mandatory;
}

.scroll-start-align {
	scroll-snap-align: start;
}
```

I think it would work to implement this in TailwindCSS, I am thinking about using it like this:

For scroll-snap-type:
- snap-both
- snap-x-mandatory
- snap-y-proximity
- snap-inline
- snap-block

_And so on_

For scroll-snap-align:
- snap-start
- snap-center
- snap-none

_And so on_

**Comment by ortuagustin** [comment]
This could be really great @adamwathan 

I've found this through this [blog](https://markus.oberlehner.net/blog/super-simple-progressively-enhanced-carousel-with-css-scroll-snap/)

It's crazy how with a few css properties you can build a carousel

Edit:
I would add [scroll-behavior](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-behavior) as well

* [scroll-snap-type docs](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-type)
* [scroll-snap-align docs](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-snap-align)

I'm personally using them like this:

```css
.scroll-snap-x-mandatory {
  scroll-snap-type: x mandatory;
}

.scroll-smooth {
  scroll-behavior: smooth;
}

.scroll-snap-start {
  scroll-snap-align: start;
}
```

I'm not too sure about the `scroll-snap-x-mandatory` class name thought

**Comment by Larsklopstra** [comment]
Bump 👍🏼

**Comment by p-lau** [comment]
I do think this will be a great addition to the main tailwind package. For the time being, I'm using [tailwind-scroll-snap](https://github.com/innocenzi/tailwindcss-scroll-snap).

**Comment by jesse-deboer** [comment]
It's there in V3! 🥳 

https://tailwindcss.com/blog/tailwindcss-v3#scroll-snap-api
